conanfile.py

```diff
@@ -0,0 +1,126 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+ARRAYFIRE_VERSION = "3.7.1"
+BINARY_INSTALLER_NAME_SUFFIX = "-1"
+BINARY_INSTALLER_NAME = f"ArrayFire-v{ARRAYFIRE_VERSION}{BINARY_INSTALLER_NAME_SUFFIX}_Linux_x86_64.sh"
+CUDA_TOOLKIT_VERSION = "10.0"
+
+class ArrayFireConan(ConanFile):
+    name = "arrayfire"
+    version = ARRAYFIRE_VERSION
+    license = "BSD"
+    author = "jacobkahn jacobkahn1@gmail.com"
+    url = "https://github.com/arrayfire/arrayfire"
+    requires = []
+    description = "ArrayFire: a general purpose GPU library"
+    topics = ("arrayfire", "gpu", "cuda", "opencl", "gpgpu",
+              "hpc", "performance", "scientific-computing")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "cpu_backend": [True, False],
+        "cuda_backend": [True, False],
+        "opencl_backend": [True, False],
+        "unified_backend": [True, False],
+        "graphics": [True, False],
+    }
+    generators = "cmake"  # unused
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration(
+                "Linux binary installer not compaible with Windows.")
+
+    def requirements(self):
+        if self.options.graphics:
+            self.requires('glfw/3.3.2@bincrafters/stable')
+
+    def _download_arrayfire(self):
+        self.af_installer_local_path = BINARY_INSTALLER_NAME
+        if not os.path.exists(self.af_installer_local_path):
+            self.output.info(
+                f"Downloading the ArrayFire {ARRAYFIRE_VERSION} binary installer...")
+            tools.download(
+                f"https://arrayfire.s3.amazonaws.com/{ARRAYFIRE_VERSION}/{BINARY_INSTALLER_NAME}", self.af_installer_local_path)
+            self.output.success(
+                f"ArrayFire {ARRAYFIRE_VERSION} binary installer successfully downloaded to {self.af_installer_local_path}")
+        else:
+            self.output.info(
+                f"ArrayFire {ARRAYFIRE_VERSION} binary installer already exists - skipping download.")
+
+    def _unpack_arrayfire(self):
+        if not os.path.exists(self.af_unpack_path):
```

Comment:
I am curious as to where is this set `af_unpack_path`

Reply:
Line 64 in `_processArrayfire` before the download or unpack.

Reply:
Ah, missed that. Thanks.